custom_connectors/oauth2/zoho_mail_connector.rb

```diff
@@ -0,0 +1,536 @@
+    },
+
+    create_task: {
+      config_fields: [
+        { name: 'Account', control_type: 'select',
+          pick_list: 'accounts', optional: true,
+          hint: 'Select the account (Zoho account/ POP account)
+          from the list of accounts available in Zoho Mail.' },
+        { name: 'From', control_type: 'select',
+          pick_list: 'fromAddress', optional: false },
+        { name: 'To', optional: false, control_type: 'email',
+          hint: 'Assignee' },
+        { name: 'Subject', optional: false },
+        { name: 'Content' }
+
+      ],
+      execute: lambda { |connection, input|
+                 account_id = if input['Account'].blank?
+                                call(:getAccountId, connection)
+                              else
+                                input['Account']
+                              end
+                 payload = {
+                   'fromAddress' => input['From'],
+                   'toAddress' => input['To'].gsub('@', '+task@'),
+                   'subject' => input['Subject'],
+                   'content' => input['Content'],
+                   'ccAddress' => input['Cc'],
+                   'bccAddress' => input['Bcc'],
+                   'mailFormat' => input['Body Type']
+                 }
+                 post("https://mail.#{connection['domain']}/"\
+                  "api/accounts/#{account_id}/messages", payload)['data']
+               },
+
+      sample_output: lambda do |_connection, input|
+        {
+          fromAddress: input['From'],
+          subject: input['Subject'],
+          toAddress: input['To'],
+          content: input['Content'],
+          ccAddress: input['Cc'],
+          bccAddress: input['Bcc']
+        }
+      end,
+
+      output_fields: lambda { |object_definitions|
+                       object_definitions['sendMailResponse']
+                     }
+    }
+
+  },
+
+  triggers: {
+
+    new_mail: {
+      type: :paging_desc,
+
+      input_fields: lambda do
+        [
+          { name: 'Account', control_type: 'select',
+            pick_list: 'accounts', optional: true,
+            hint: 'Select the account (Zoho account/ POP account)
+             from the list of accounts available in Zoho Mail.' }
+        ]
+      end,
+      poll: lambda do |connection, input, page|
+        limit = 50
+        page ||= 0
+        offset = (limit * page) + 1
+        account_id = if input['Account'].blank?
+                       call(:getAccountId, connection)
+                     else
+                       input['Account']
+                     end
+
+        response = get("https://mail.#{connection['domain']}"\
+          "/api/accounts/#{account_id}/messages/search")
+                   .params(searchKey: 'newMails',
+                           limit: limit,
+                           includeto: 'true',
+                           start: offset,
+                           receivedTime: Time.now.to_i * 1000 -
```

Comment:
Use the `input['since']` here instead of fixing the `receivedTime` to some constant value.
Something like this - 
`receivedTime: input['since'].to_i`